webapp/src/auth.ts

```diff
@@ -256,6 +263,43 @@ async function fetchUserAsync() {
     }
 }
 
+
+export async function updateUserPreferencesAsync(newPref: Partial<UserPreferences>) {
+    state_.preferences = { ...(state_.preferences || {}), ...newPref }
+    data.invalidate("user-pref");
+
+    if (!loggedIn()) { return; }
+
+    // If the user is logged in, save to cloud
+    const result = await apiAsync<UserPreferences>('/api/user/preferences', newPref);
+    if (result.success) {
+        console.log("updating userpref from POST")
+        // Set user profile from returned value
+        state_.preferences = { ...state_.preferences, ...result.resp }
+        data.invalidate("user-pref");
```

Comment:
>"user-pref" [](start = 24, length = 11)

Duplicating "user-pref" everywhere can lead to problems later. I think it would be better to define a const for it at the top of this file.